datarobot_batch_scoring/network/base_network_worker.py

```diff
@@ -0,0 +1,209 @@
+import collections
+import logging
+import os
+import sys
+
+from datarobot_batch_scoring.consts import (WriterQueueMsg, Batch,
+                                            ProgressQueueMsg)
+from datarobot_batch_scoring.reader import (fast_to_csv_chunk,
+                                            slow_to_csv_chunk)
+from datarobot_batch_scoring.utils import compress, get_rusage, Worker
+
+
+logger = logging.getLogger(__name__)
+
+
+FakeResponse = collections.namedtuple('FakeResponse', 'status_code, text')
+
+
+class BaseNetworkWorker(Worker):
+    """A network worker
+
+    Work for the worker is read off of the network_queue; failures are put into
+    the network_deque from which they are also read and sometimes retried.
+
+    Successfully returned responses are put into the writer_queue for
+    processing by another worker.
+
+    Occasional progress updates are occasionally put into the progress_queue
+    for processing by another worker
+
+    A properly functioning Worker implementation must read data from the
+    network_queue until it finds the SENTINEL message, which is the indicator
```

Comment:
Sentinel is no longer used really as it does not work with multiple workers.

Actually worker just indicates idle state when no new requests are available in input queues, and main process monitors states of all workers and goes thru app state.

At first it waits for successful completion of Reader, after that it waits for all network workers go Idle, and waits for writer after that.